node/src/components/storage.rs

```diff
@@ -1796,6 +1757,71 @@ impl Storage {
         let message = Message::new_get_response_from_serialized(<T as Item>::TAG, shared);
         Ok(effect_builder.send_message(sender, message).ignore())
     }
+
+    fn get_available_block_range(&self) -> AvailableBlockRange {
+        let high = self
+            .block_height_index
+            .keys()
+            .last()
+            .copied()
+            .unwrap_or_default();
+        let low = self.lowest_available_block_height;
+        match AvailableBlockRange::new(low, high) {
+            Ok(range) => range,
+            Err(error) => {
+                error!(%error);
+                AvailableBlockRange::default()
+            }
+        }
+    }
+
+    fn update_lowest_available_block_height(
+        &mut self,
+        new_height: u64,
+    ) -> Result<(), FatalStorageError> {
+        // We should update the value if
+        // * it wasn't already stored, or
+        // * the new value is lower than the available range at startup, or
+        // * the new value is 2 or more higher than (after pruning due to hard-reset) the available
```

Comment:
nitpick: maybe it's worth including an explanation why is that a problem since it may not be obvious to the reader.  I.e. just write that there would be a single-block gap in the sequence where we might not have the Global State.